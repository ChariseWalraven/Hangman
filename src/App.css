@import url('https://fonts.googleapis.com/css?family=Caveat');

.App {
  text-align: center;
}

.Hangman {
  position: absolute;
  width: 100%;
  top: 20%;
}