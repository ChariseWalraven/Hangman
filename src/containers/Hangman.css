.Settings__container {
  position: fixed;
  top: 2.5%;
  right: 2.5%;
}

.Settings__container > div > svg {
  color: rgba(0, 0, 0, 0.74);
}
