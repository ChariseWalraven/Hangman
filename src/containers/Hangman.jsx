import React, { Component } from 'react'
import { connect } from 'react-redux'
import { guess, updateGuess, resetGuess, newWord, resetState, deleteLastLetter } from '../actions/game'
import Game from '../lib/game'
import { Typography } from '@material-ui/core'
import Word from '../components/Word'

class Hangman extends Component {
  componentWillMount() {
    this.props.newWord()
    let history = [];
    let guess = [];
    document.addEventListener('keydown', e => {
      const yes = 'aAbBcCdDeEfFgGhHiIjJkKlLmMnNoOpPqQrRsStTuUvVwWxXyYzZ'.split('')
      // const no = [16, 192, 9, 20]
      if (e.keyCode === 8) {
        // dispatch action to remove last bit of guess
        guess = guess.slice(0, guess.length - 1)
        this.props.deleteLastLetter()
      }
      if (yes.includes(e.key)) {
        //if guess is in the array, push the guess, and the history
        guess.push(e.key)
        history.push(e.key)
      }
      if (e.key === 'Enter') {
        if (this.props.guessesRemaining > 0) {
          this.handleSubmit(guess, history)
        }
        else {
          //reset game
        }
      }
    })
  }

  handleSubmit = (guess, history) => {
    // update store with wrong guess count
    if (this.props.guessesRemaining > 0) {
      let wrong = this.props.guessesRemaining - Game.wrongGuessCount(this.props.word, guess)
      this.props.updateGuess(wrong, history)
    }
    else {
      // reset guesses remaining and get a new letter
      this.props.resetGuess()
      console.log(this.props.userGuess)
    }

  }

  handleClick = () => {
    // trigger notification
    this.props.newWord()

  }

  render() {
    return (
      <div className='Hangman'>
        <Typography variant='display1'>
          Hit Enter to Submit your guess!
        </Typography>
        <Typography variant='display1'>
          Your guess:
          {" " + this.props.userGuess.join("")}
        </Typography>
        {
          this.props.display !== null ?
            <Word props={this.props} handleClick={this.handleClick} />
            : null
        }
        <Typography variant='display1' component='div'>
          Guesses remaining:
          <Typography component='span' variant='display1' color={'error'}>
            {this.props.guessesRemaining}
          </Typography>
        </Typography>
      </div>
    )
  }
}


const mapStateToProps = (state) => ({
  guessesRemaining: state.guess.guessesRemaining,
  word: state.guess.word,
  userGuess: state.guess.guess,
  currentLetter: state.guess.currentLetter,
  display: state.guess.display ? state.guess.display : null
})

export default connect(mapStateToProps, {
  guess, updateGuess, resetGuess, newWord, resetState, deleteLastLetter
})(Hangman)
