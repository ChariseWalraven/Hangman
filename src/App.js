import React, { Component } from 'react';
import Hangman from './components/Hangman'
import Nav from './containers/Nav'
import { MuiThemeProvider, createMuiTheme, withStyles } from '@material-ui/core';
import { blue, teal, red } from '@material-ui/core/colors'
import './App.css';
import Footer from './containers/Footer';
import ErrorBoundary from './ErrorBoundary'

const theme = createMuiTheme({
  typography: {
    fontFamily: [
      'Indie Flower',
      'cursive'
    ].join(','),
    htmlFontSize: 16,
    display4: {
      lineHeight: '1.9em'
    },
    display1: {
      lineHeight: '1.8em'
    }
  },
  palette: {
    primary: blue,
    secondary: teal,
    error: red,
  },
  spacing: {
    unit: 8
  }
})

const styles = theme => ({
  root: {
    display: 'flexGrow',
  }
})

class App extends Component {
  render() {
    return (
      <MuiThemeProvider theme={theme}>
        <div className="App">
          <Nav theme={theme} />
          <ErrorBoundary>
            <Hangman theme={theme} />
          </ErrorBoundary>
          <Footer theme={theme} />
        </div>
      </MuiThemeProvider>
    );
  }
}

export default withStyles(styles)(App);
