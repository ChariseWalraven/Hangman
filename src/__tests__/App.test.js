import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import Hangman from '../containers/Hangman'


describe('<App>', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(
      <App />
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  });
  it('should render a Hangman component', () => {
    const rendered = renderer.create(
      <Hangman />
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
