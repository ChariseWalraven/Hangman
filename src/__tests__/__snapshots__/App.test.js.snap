// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> renders without crashing 1`] = `
<div
  className="App"
>
  <div
    id="Header"
  >
    <div>
      <h1
        className="MuiTypography-root-2 MuiTypography-display4-3 MuiTypography-colorError-25 MuiTypography-alignCenter-15"
      >
        Hangman
      </h1>
    </div>
  </div>
  <div
    className="Hangman"
  >
    <h1
      className="MuiTypography-root-2 MuiTypography-display1-6"
    >
      Hit Enter to Submit your guess!
    </h1>
    <h1
      className="MuiTypography-root-2 MuiTypography-display1-6"
    >
      Your guess:
       
    </h1>
    <div
      className="MuiTypography-root-2 MuiTypography-display1-6"
    >
      <span>
        _______
      </span>
      <svg
        aria-describedby={undefined}
        aria-hidden="true"
        className="MuiSvgIcon-root-34"
        color={undefined}
        focusable="false"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M12 6v3l4-4-4-4v3c-4.42 0-8 3.58-8 8 0 1.57.46 3.03 1.24 4.26L6.7 14.8c-.45-.83-.7-1.79-.7-2.8 0-3.31 2.69-6 6-6zm6.76 1.74L17.3 9.2c.44.84.7 1.79.7 2.8 0 3.31-2.69 6-6 6v-3l-4 4 4 4v-3c4.42 0 8-3.58 8-8 0-1.57-.46-3.03-1.24-4.26z"
          />
        </g>
      </svg>
    </div>
    <div
      className="MuiTypography-root-2 MuiTypography-display1-6"
    >
      Guesses remaining:
      <span
        className="MuiTypography-root-2 MuiTypography-display1-6 MuiTypography-colorError-25"
      >
        7
      </span>
    </div>
  </div>
  <h1>
    Something went wrong.
  </h1>
  <div
    className="MuiPaper-root-41 MuiPaper-elevation4-47 MuiPaper-rounded-42 Footer-root-40"
    id="Footer"
  >
    <div
      className="MuiGrid-container-68 MuiGrid-spacing-xs-16-90 Footer-root-40"
    >
      <div
        className="MuiGrid-item-69 MuiGrid-grid-xs-12-107"
      >
        <div
          className="MuiGrid-container-68 MuiGrid-spacing-xs-16-90 MuiGrid-align-items-xs-flex-end-78 MuiGrid-justify-xs-center-85"
        >
          <a
            className="MuiTypography-root-2 MuiTypography-subheading-9"
            href="#"
          >
            Copyright Charise Walraven 2018
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
